org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/migration/AbstractRemoveDoEntityContributionValueMigrationHandlerTest.java

```diff
@@ -0,0 +1,276 @@
+    IDoEntity expectedUnknownContribution2 = BEANS.get(DoEntity.class);
+    expectedUnknownContribution2.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.MyOtherTestContribution");
+    expectedUnknownContribution2.put(DoStructureMigrationHelper.TYPE_VERSION_ATTRIBUTE_NAME, AlfaFixture_1.VERSION.unwrap());
+    expectedUnknownContribution2.put("otherTestValue", "myOtherTestValue");
+
+    // build the DO manually, because we want to add unknown/invalid contributions
+    IDoEntity expectedDataObject = BEANS.get(DoEntity.class);
+    expectedDataObject.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.SimpleFixture");
+    expectedDataObject.put("name1", "myName1");
+    expectedDataObject.putList("_contributions", CollectionUtility.arrayList(expectedExistingContribution1, expectedExistingContribution2, expectedUnknownContribution1, expectedUnknownContribution2));
+
+    // serialize and deserialize the DO, so that we get a SimpleFixtureDo instead of a DoEntity
+    IDoEntity expected = s_dataObjectMapper.readValue(s_dataObjectMapper.writeValue(expectedDataObject), IDoEntity.class);
+
+    testMigration(createDataObjectWithMultipleContributions(), expected);
+  }
+
+  protected void testMigration(IDoEntity value, IDoEntity expected) {
+    String json = s_dataObjectMapper.writeValue(value);
+    DataObjectMigrationContext ctx = BEANS.get(DataObjectMigrationContext.class).putGlobal(BEANS.get(DoValueMigrationIdsContextData.class).withAppliedValueMigrationIds(getAppliedValueMigrations()));
+    DataObjectMigratorResult<IDoEntity> result = BEANS.get(DataObjectMigrator.class).migrateDataObject(ctx, json, IDoEntity.class);
+    assertTrue("Data object migration didn't change the data object", result.isChanged());
+    assertEquals(expected, result.getDataObject());
+  }
+
+  /**
+   * @return the set of all available value migration IDs, except the migrations to be tested. This is to make sure that no other value
+   * migrations than the expected ones will be applied.
+   */
+  protected Set<DoValueMigrationId> getAppliedValueMigrations() {
+    return BEANS.get(DataObjectMigrationInventory.class).getValueMigrationHandlers().stream()
+        .map(IDoValueMigrationHandler::id)
+        .filter(id -> !MyTestContributionRemoveDoEntityContributionValueMigrationHandler.ID.equals(id))
+        .filter(id -> !FirstSimpleContributionFixtureRemoveDoEntityContributionValueMigrationHandler.ID.equals(id))
+        .collect(Collectors.toSet());
+  }
+
+  protected SimpleFixtureDo createSimpleFixtureDo() {
+    return BEANS.get(SimpleFixtureDo.class)
+        .withName1("myName1");
+  }
+
+  protected SimpleFixtureDo createDataObjectWithExistingContribution() {
+    SimpleFixtureDo fixture = createSimpleFixtureDo();
+    fixture.contribution(FirstSimpleContributionFixtureDo.class)
+        .withFirstValue("myFirstValue");
+    return fixture;
+  }
+
+  protected IDoEntity createDataObjectWithUnknownContribution() {
```

Comment:
This method should return a typed data object (i.e. a SimpleFixtureDo). Else the value migration handlers will not handle the data object